django_mongodb_backend/gis/operations.py

```diff
@@ -1,11 +1,110 @@
+import warnings
+
 from django.contrib.gis import geos
 from django.contrib.gis.db import models
 from django.contrib.gis.db.backends.base.operations import BaseSpatialOperations
+from django.contrib.gis.measure import Distance
+from django.db.backends.base.operations import BaseDatabaseOperations
 
 from .adapter import Adapter
+from .utils import SpatialOperator
+
+
+def _gis_within_operator(field, value, op=None, params=None):
+    print(f"Within value: {value}")
+    return {
+        field: {
+            "$geoWithin": {
+                "$geometry": {
+                    "type": value["type"],
+                    "coordinates": value["coordinates"],
+                }
+            }
+        }
+    }
+
+
+def _gis_intersects_operator(field, value, op=None, params=None):
+    return {
+        field: {
+            "$geoIntersects": {
+                "$geometry": {
+                    "type": value["type"],
+                    "coordinates": value["coordinates"],
+                }
+            }
+        }
+    }
+
+
+def _gis_disjoint_operator(field, value, op=None, params=None):
+    return {
+        field: {
+            "$not": {
+                "$geoIntersects": {
+                    "$geometry": {
+                        "type": value["type"],
+                        "coordinates": value["coordinates"],
+                    }
+                }
+            }
+        }
+    }
+
 
+def _gis_contains_operator(field, value, op=None, params=None):
+    value_type = value["type"]
+    if value_type != "Point":
+        warnings.warn(
+            "MongoDB does not support strict contains on non-Point query geometries. Results will be for intersection."
+        )
```

Comment:
I tend to think we should use `NotSupportedError` rather than giving some result that doesn't match other backends.